hibernate-core/src/main/java/org/hibernate/query/internal/ParameterMetadataImpl.java

```diff
@@ -68,24 +69,20 @@ public ParameterMetadataImpl(Map<QueryParameterImplementor<?>, List<SqmParameter
 					labels = new HashSet<>();
 				}
 				labels.add( queryParameter.getPosition() );
+				queryParametersByLabels.put( queryParameter.getPosition(), queryParameter );
 			}
 			else if ( queryParameter.getName() != null ) {
 				if ( names == null ) {
 					names = new HashSet<>();
 				}
-				names.add( queryParameter.getName() );
+				final String name = queryParameter.getName();
+				names.add( name );
```

Comment:
Looks like this set and the `labels` set are no longer used and can just be removed.